controller/proxy-injector/webhook_test.go

```diff
@@ -153,6 +154,158 @@ func TestGetPatch(t *testing.T) {
+				expectedLabelValue: "emoji-deploy",
+			},
+			{
+				k8sConfigs: []string{},
+				pod: `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: emoji-svc
+  namespace: emojivoto
+  ownerReferences:                                                      
+  - apiVersion: v1
+    kind: ReplicationController                                                    
+    name: emoji-rc`,
+				expectedLabelKey:   pkgK8s.ProxyReplicationControllerLabel,
+				expectedLabelValue: "emoji-rc",
+			},
+			{
+				k8sConfigs: []string{`
+apiVersion: apps/v1beta2
+kind: ReplicaSet
+metadata:
+  name: emoji-rs
+  namespace: emojivoto
+  ownerReferences:
+  - apiVersion: apps/v1beta2
+    kind: Deployment
+    name: emoji-deploy`,
+				},
+				pod: `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: emoji-svc
+  namespace: emojivoto
+  ownerReferences:                                                      
+  - apiVersion: apps/v1beta2
+    kind: ReplicaSet                                                    
+    name: emoji-rs`,
+				expectedLabelKey:   pkgK8s.ProxyDeploymentLabel,
+				expectedLabelValue: "emoji-deploy",
+			},
+			{
+				k8sConfigs: []string{},
+				pod: `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: emoji-svc
+  namespace: emojivoto
+  ownerReferences:                                                      
+  - apiVersion: batch/v1
+    kind: Job                                                
+    name: emoji-job`,
+				expectedLabelKey:   pkgK8s.ProxyJobLabel,
+				expectedLabelValue: "emoji-job",
+			},
+			{
+				k8sConfigs: []string{},
+				pod: `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: emoji-svc
+  namespace: emojivoto
+  ownerReferences:                                                      
+  - apiVersion: apps/v1
+    kind: DaemonSet                                         
+    name: emoji-ds`,
+				expectedLabelKey:   pkgK8s.ProxyDaemonSetLabel,
+				expectedLabelValue: "emoji-ds",
+			},
+			{
+				k8sConfigs: []string{},
+				pod: `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: emoji-svc
+  namespace: emojivoto
+  ownerReferences:                                                      
+  - apiVersion: apps/v1
+    kind: StatefulSet                                         
+    name: emoji-sts`,
+				expectedLabelKey:   pkgK8s.ProxyStatefulSetLabel,
+				expectedLabelValue: "emoji-sts",
+			},
+		}
+
+		for _, tt := range testCases {
+			k8sConfigs := append(tt.k8sConfigs, tt.pod)
+			k8sAPI, err := k8s.NewFakeAPI(k8sConfigs...)
+			if err != nil {
+				t.Fatalf("Error instantiating client: %s", err)
+			}
+			k8sAPI.Sync()
+
+			webhook, err := NewWebhook(k8sAPI, "emojivoto", false)
+			if err != nil {
+				t.Fatalf("Error instantiating Webhook: %s", err)
+			}
+
+			b := []byte(tt.pod)
+
+			conf := confNsEnabled()
+			nonEmpty, err := conf.ParseMeta(b, "")
+			if err != nil {
+				t.Fatal(err)
+			}
+			if !nonEmpty {
+				t.Fatalf("Unexpected empty result from ParseMeta()")
+			}
+
+			if err := conf.Parse(b); err != nil {
```

Comment:
I don't think we should export this function just for the test. Can you use `inject.ParseMetaAndYaml()` here?